squirrels/_auth.py

```
@@ -468,6 +411,298 @@ def get_user_from_token(self, token: str | None) -> tuple[RegisteredUser | None,
+                response.raise_for_status()
+                user_info = response.json()
+                if isinstance(user_info, dict) and user_info:
+                    return user_info
+            except Exception:
+                # Fall back to introspection if available
+                pass
+
+        introspection_endpoint = metadata.get("introspection_endpoint")
+        if isinstance(introspection_endpoint, str) and introspection_endpoint:
+            try:
+                response = requests.post(
```

Ruff (0.14.13)

Probable use of `requests` call without timeout

(S113)